android/java/org/chromium/chrome/browser/vpn/BraveVpnConfirmDialogFragment.java

```diff
@@ -0,0 +1,109 @@
+import android.view.WindowManager;
+import android.widget.LinearLayout;
+
+import androidx.fragment.app.DialogFragment;
+
+import com.airbnb.lottie.LottieAnimationView;
+
+import org.chromium.chrome.R;
+import org.chromium.chrome.browser.util.ConfigurationUtils;
+import org.chromium.ui.base.DeviceFormFactor;
+
+public class BraveVpnConfirmDialogFragment extends DialogFragment {
+    private LottieAnimationView mAnimatedView;
+
+    @Override
+    public void onConfigurationChanged(Configuration newConfig) {
+        super.onConfigurationChanged(newConfig);
+        setDialogParams();
+    }
+
+    @Override
+    public void onResume() {
+        super.onResume();
+        getDialog().setOnKeyListener(new DialogInterface.OnKeyListener() {
+            @Override
+            public boolean onKey(android.content.DialogInterface dialog, int keyCode,
+                    android.view.KeyEvent event) {
+                if ((keyCode == android.view.KeyEvent.KEYCODE_BACK)) {
+                    dismiss();
+                    return true;
+                } else
+                    return false;
+            }
+        });
+        setDialogParams();
+    }
+
+    @Override
+    public View onCreateView(
+            LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
+        return inflater.inflate(R.layout.fragment_brave_vpn_confirm_dialog, container, false);
+    }
+
+    @Override
+    public void onViewCreated(View view, Bundle savedInstanceState) {
+        mAnimatedView = view.findViewById(R.id.bg_image);
+        mAnimatedView.setAnimation("brave_vpn_confirm.json");
+        mAnimatedView.addAnimatorListener(new Animator.AnimatorListener() {
+            @Override
+            public void onAnimationStart(Animator animation) {}
+
+            @Override
+            public void onAnimationEnd(Animator animation) {
+                try {
+                    Thread.sleep(300);
+                } catch (InterruptedException e) {
+                    e.printStackTrace();
+                }
+                getActivity().finish();
+            }
+
+            @Override
+            public void onAnimationCancel(Animator animation) {}
+
+            @Override
+            public void onAnimationRepeat(Animator animation) {}
+        });
+        mAnimatedView.playAnimation();
+    }
+
+    private void setDialogParams() {
+        DisplayMetrics displayMetrics = new DisplayMetrics();
+        getActivity().getWindowManager().getDefaultDisplay().getMetrics(displayMetrics);
+        int mDeviceHeight = displayMetrics.heightPixels;
+        int mDeviceWidth = displayMetrics.widthPixels;
+
+        WindowManager.LayoutParams params = getDialog().getWindow().getAttributes();
+        boolean isTablet = DeviceFormFactor.isNonMultiDisplayContextOnTablet(getActivity());
+        boolean isLandscape = ConfigurationUtils.isLandscape(getActivity());
+        if (isTablet) {
+            params.width = (int) (0.5 * mDeviceWidth);
```

Comment:
should there be landscape/portrait widths for tablet too?

Reply:
@alexsafe usually for custom dialogs, we keep the width same in landscape/portrait